recipe-page/recipe-page.js

```diff
@@ -1,32 +1,86 @@
-let recipePopup = document.getElementById('popup1')
-let recipeButton = document.getElementById('recipe-details-button')
-let span1 = document.getElementsByClassName('close')[0]
+// API Section 
+let cardID = localStorage.getItem("CardId");
+let mealsArray = {};
+let recipeInfo = document.querySelector(".recipe-info");
+const recipePicture = document.querySelector(".recipe-picture");
+const recipeDetails = document.querySelector(".recipe-details");
+const ingredientsDetails = document.querySelector(".ingredients-details");
+const pageTitle = document.getElementsByTagName("title")[0]
 
+console.log(cardID);
+fetch("https://www.themealdb.com/api/json/v1/1/lookup.php?i=" + cardID)
+  .then((res) => {
+    return res.json();
+  })
+  .then((data) => {
+    mealsArray = data.meals[0];
+    let title = document.createElement("h1");
+    console.log("name", mealsArray.strMeal);
+    title.innerText = pageTitle.textContent = mealsArray.strMeal;
+    console.log(pageTitle)
+    title.classList.add("recipe-name");
+    recipeInfo.appendChild(title);
+    recipePicture.innerHTML = ` <a href = ${data.meals[0].strYoutube}>  <img class="recipe-img" src=${data.meals[0].strMealThumb} alt="No picture available"/> </a>
+    <h4 class="notice">Click on the picture for an illustrative video</h4>`; //Youtube Video
```

Comment:
using innerHTML is considered a bad practice, therefore, it is not recommended to use it

use createElement to create a new element ( image, a, h4) and use appendChild to append the element with the parent element